app/components/card_component.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+class CardComponent < ApplicationComponent
+  renders_one :header
+  renders_one :footer
+
+  erb_template <<~ERB
+    <div class="panel panel-<%= @state %>">
+
+      <% if header? %>
+      <div class="panel-heading">
+        <%= header %>
+      </div>
+      <% end %>
+
+      <div class="panel-body">
+        <%= content %>
+      </div>
+
+      <% if footer? %>
+      <div class="panel-footer">
+        <%= footer %>
+      </div>
+      <% end %>
+    </div>
+  ERB
+
+  def initialize(state = :default)
+    @state = state
```

Comment:
```suggestion
    @type = type
```

Perfer type over state.

Also add list of available types in a constant. And if value is not in the list then raise an `ArgumentError` exception:

```rb
raise ArgumentError, "#{type} is not a valid type" unless TYPES.key?(type)
```